cmd/soroban-rpc/internal/db/cursor.go

```diff
@@ -27,6 +27,13 @@ type Cursor struct {
 	Event uint32
 }
 
+type CursorRange struct {
+	// Start defines the (inclusive) start of the range.
+	Start Cursor
+	// End defines the (exclusive) end of the range.
```

Comment:
Can you double-check that this is the same behavior as for the other paginated endpoints? I thought we started the next page of reads at +1 the given start cursor (cc @aditya1702 for `getTransactions` context)

Reply:
Just to confirm the concern, you meant increment cursor id by 1 for txn/events if cursor is present in request?

Pagination for events is like this:
```

if request.Pagination.Cursor != nil {
			start = *request.Pagination.Cursor
			// increment event index because, when paginating,
			// we start with the item right after the cursor
			start.Event++
		}
```

Reply:
Equal to what we have for Transaction:
```

if request.Pagination != nil {
		if request.Pagination.Cursor != "" {
			cursorInt, err := strconv.ParseInt(request.Pagination.Cursor, 10, 64)
			if err != nil {
				return toid.ID{}, 0, &jrpc2.Error{
					Code:    jrpc2.InvalidParams,
					Message: err.Error(),
				}
			}
			*start = toid.Parse(cursorInt)
			// increment tx index because, when paginating,
			// we start with the item right after the cursor
			start.TransactionOrder++
		}
		if request.Pagination.Limit > 0 {
			limit = request.Pagination.Limit
		}
	}
```

Reply:
cc @aditya1702  for another look and then I will resolve this if all looks good.

Reply:
Right, so doesn't that mean `Start` is not inclusive? So the comment is incorrect.

Reply:
I think this comment was there in the context of DB operation afaik see below query. Also just to add this is existing cursor logic for events in memory too.

our SQL query is like this:

```
Where(sq.GtOrEq{"id": cursorRange.Start.String()}).
Where(sq.Lt{"id": cursorRange.End.String()}).
```